packages/back-end/src/models/ApiKeyModel.ts

```diff
@@ -28,7 +29,13 @@ const apiKeySchema = new mongoose.Schema({
 
 type ApiKeyDocument = mongoose.Document & ApiKeyInterface;
 
-const ApiKeyModel = mongoose.model<ApiKeyDocument>("ApiKey", apiKeySchema);
+const ApiKeyModel = mongoose.model<ApiKeyInterface>("ApiKey", apiKeySchema);
+
+const toInterface = (doc: ApiKeyDocument): ApiKeyInterface =>
+  omit(
+    doc.toJSON<ApiKeyDocument>({ flattenMaps: true }),
```

Comment:
We need to start passing the document type to `.toJSON<T>()`. Passing our interface did not appear to work with Mongoose 7—there was an error about missing property `id` in all models.

Information about the `flattenMaps` option can be found here: https://mongoosejs.com/docs/api/map.html#MongooseMap.prototype.toJSON() – I don't think this is relevant for our models as we don't appear to be using any `Map` instances in our models, and if I'm reading the types correctly, it seems like `true` is the default option, so we may want to not include any options.